### PR TITLE
Implemented backend & sdk for new `HttpService` aggregation

### PR DESCRIPTION
**Pull request by gammelalf**

### CODE REVIEW

File: kraken/src/models/manual/mod.rs [hunk 1]
```diff
@@ -147,3 +147,46 @@ pub struct ManualDomain {
...
+    ///
+    /// (Should default to "/")
+    #[rorm(max_length = 1024)]
+    pub base_path: String,
+
+    /// Is this a https service?
+    pub tls: bool,
+
+    /// Does this service require sni?
+    pub sni_require: bool,
```
Comment (myOmikron) [inline kraken/src/models/manual/mod.rs:180]: typo, `sni_required`

File: kraken/src/modules/aggregator/mod.rs [hunk 2]
```diff
@@ -45,6 +46,16 @@ struct ServiceAggregationData {
...
 }
+struct HttpServiceAggregationData {
+    workspace: Uuid,
+    name: String,
+    host: Uuid,
+    port: Uuid,
+    domain: Option<Uuid>,
+    base_path: String,
+    tls: bool,
+    sni_required: bool,
```
Comment (myOmikron) [inline kraken/src/modules/aggregator/mod.rs:57]: typo

File: kraken/src/models/manual/operations.rs [hunk 3]
```diff
@@ -386,3 +390,146 @@ impl ManualService {
...
+        workspace: Uuid,
+        user: Uuid,
+        name: String,
+        domain: Option<String>,
+        ip_addr: IpAddr,
+        port: NonZeroU16,
+        base_path: String,
+        tls: bool,
+        sni_require: bool,
+    ) -> Result<Uuid, rorm::Error> {
```
Comment (myOmikron) [inline kraken/src/models/manual/operations.rs:428]: Protocol is missing

Comment (myOmikron) [reply kraken/src/models/manual/operations.rs:428]: in case we are dealing with quick, we mustn't assume tcp

**Review by myOmikron — CHANGES_REQUESTED** [review]

File: kraken/src/modules/aggregator/mod.rs (hunk 2 above)
Comment (gammelalf) [reply kraken/src/modules/aggregator/mod.rs:57]: Where?

File: kraken/src/models/manual/operations.rs (hunk 3 above)
Comment (gammelalf) [reply kraken/src/models/manual/operations.rs:428]: Which option would you prefer?
1. a boolean flag named something like `is_tcp` or `is_udp`
2. using `PortProtocol` which contains other variants besides TCP and UDP
3. create a new enum
This choices concerns the schema exposed to the API, the aggregator's method and the manual insert model

File: kraken/src/modules/aggregator/mod.rs (hunk 2 above)
Comment (myOmikron) [reply kraken/src/modules/aggregator/mod.rs:57]: should be fixed by force pushing ^^

File: kraken/src/models/manual/operations.rs (hunk 3 above)
Comment (myOmikron) [reply kraken/src/models/manual/operations.rs:428]: No strong opinion between 2 and 3, no fan of 1

Comment (gammelalf) [reply kraken/src/models/manual/operations.rs:430]: went for 2 now

File: kraken/src/api/handler/http_services/handler.rs [hunk 4]
```diff
@@ -0,0 +1,807 @@
...
+        port,
+        port_protocol,
+        certainty,
+        base_path,
+        tls,
+        sni_require,
+    } = req.into_inner();
+    let PathUuid { uuid: workspace } = path.into_inner();
+
+    Ok(Json(UuidResponse {
```
Comment (myOmikron) [inline kraken/src/api/handler/http_services/handler.rs:446]: Check if user is member or owner

**Review by myOmikron — CHANGES_REQUESTED** [review]

**Review by myOmikron — APPROVED** [review]

**Pull request merged**